executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Chain.hs

```diff
@@ -113,105 +110,99 @@ instance ( HashAlgorithm hashAlgo
-                 , closedVotingPeriods = Map.empty
-                 , transactionsSt = mempty
-                 }
+       IRC Env { headerEnv = Header.Env{ Header.initialSlot = initsl}
+               } <- judgmentContext
+       pure $! St { headerSt
+                      = Header.St { Header.currentSlot = initsl
+                                  , Header.hupdateSt
+                                      = Hupdate.St { Hupdate.wrsips
+                                                      = Map.empty
+                                                   , Hupdate.asips
+                                                      = Map.empty
+                                                   }
+                                  }
+                  , bodySt
+                      = Body.St
+                        { Body.wrsips = Map.empty
+                        , Body.transactionSt
+                            = Transaction.St
+                                 { Transaction.wrsips = Map.empty
+                                 , Transaction.utxoSt = mempty
+                                 , Transaction.updateSt
+                                      = Update.St { Update.wrsips = Map.empty
+                                                  , Update.ideationSt
+                                                      = Ideation.St
+                                                        { Ideation.subsips
+                                                            = Set.empty
+                                                        , Ideation.wssips
+                                                            = Map.empty
+                                                        , Ideation.wrsips
+                                                            = Map.empty
+                                                        , Ideation.ballots
+                                                            = Map.empty
+                                                        , Ideation.voteResultSIPs
+                                                            = Map.empty
+                                                        }
+                                                  , Update.implementationSt
+                                                      = Implementation.St ()
+                                                  }
+                                 }
+                        }
+                  }
     ]
 
   transitionRules = [
     do
-      TRC ( Env { maximumBlockSize, transactionsEnv }
-          , St  { currentSlot
-                , openVotingPeriods
-                , closedVotingPeriods
-                , transactionsSt =
-                    Transaction.St { -- Transaction.openVotingPeriods
-                                     Transaction.utxoSt
-                                   , Transaction.updateSt
-                                   }
+      TRC ( Env { maximumBlockSize, headerEnv, bodyEnv = Body.Env{Body.transactionEnv} }
```

Comment:
I'm a bit confused by this `transactionEnv` here. It seems to contain things that should come from the `CHAIN` state:

```haskell
  Env { currentSlot :: !Slot
      , asips :: !(Map (Data.SIPHash hashAlgo) Slot)
      -- ^ When a SIP will not be active any more
      -- (i.e., end of open for voting period)
      , updatesEnv :: !(Environment (UPDATES hashAlgo))
      -- ^ Environment of the child STS (UPDATES)
      , utxoEnv :: !(Environment UTXO)
      -- ^ Environment of the child STS (UTXO)
```

Reply:
Furthermore, the body environment contains `asips` but also a transaction environment that also contains `asips`.